### PR TITLE
Fix schema migration with DeduplicateAction

### PR DESCRIPTION
**Pull request by MarkusRothSBB**
### What changes are included in the pull request?
-Refactor DeduplicateAction for better testability
-Fix bug regarding schema migration and DeduplicateAction
-Introduce new test helpers for easier testing

### Why are the changes needed?
I am still working on the new test helpers as discussed. In the end, I think they should not live in test but in main, so that applications using SDL can use the test helpers when writing tests for their applications.
However, I added them here because I needed to write tests to fix a bug in the DeduplicateAction regarding schema evolution, and I think they allow for much easier to follow tests than the ones present.

I need the bugfix soon, so if we want to postpone adding the testhelpers until they are more refined, that's fine with me. In that case, I'll modify the pull request to just include the bugfix with no tests, and we will re-add them when the test helpers are complete.

### CODE REVIEW

File: src/test/scala/io/smartdatalake/testutils/DataFrameTestHelper.scala [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+  implicit def valueToTypedValue[T](value: T): TypedValue = value match {
+    case (value: Any, dataType: DataType) => TypedValue(value, dataType)
+    case (null, dataType: DataType) => TypedValue(null, dataType)
+    case string: String => TypedValue(string, StringType)
+    case int: Int => TypedValue(int, IntegerType)
+    case bool: Boolean => TypedValue(bool, BooleanType)
+    case _ => throw new Exception("Unable to convert to TypedValue")
+  }
+
+  def assertSchemasEqual(expected: StructType, actual: StructType): Unit = {
```
Comment (zzeekk) [inline src/test/scala/io/smartdatalake/testutils/DataFrameTestHelper.scala:170]: In assertDataFramesEqual we have another implementation of schema comparision. Could we merge them?
I would prefere the one from assertDataFramesEqual.

File: src/main/scala/io/smartdatalake/workflow/action/DeduplicateAction.scala [hunk 2]
```diff
@@ -93,26 +93,55 @@ case class DeduplicateAction(override val id: ActionObjectId,
...
     } else enhancedDf
   }
 
+  /**
+   * The techincal "captured" column needs to be the last column in the DataFrame so that the union in the deduplicate method words
+   * In Spark, unions do not work by column name but by column index, so the column order of the dataframes to be unioned needs to be equal
+   * @param df A DataFrame
+   * @return A DataFrame where the techinical "captured" column is the last column
+   */
+  def moveCapturedColumnToEnd(df: DataFrame) : DataFrame = {
```
Comment (zzeekk) [inline src/main/scala/io/smartdatalake/workflow/action/DeduplicateAction.scala:140]: I also came accross this problem when working on #182. There is a better solution by using df.unionByName in deduplicate(...). Shouldn't we go directly for that?

**Review by zzeekk — COMMENTED** [review]
Thanks @MarkusRothSBB,

I have made some comments, and we would also need to consolidate DataFrameTestHelper and TestUtils. But I suggest to do this later in another PR.

**Comment by MarkusRothSBB** [comment]
I can't push updates to this PR because I chose the wrong branch name. Closing this to re-open with a better branch name.

**Pull request closed without merging**